app/models/tagging.rb

```diff
@@ -3,4 +3,5 @@
 class Tagging < ApplicationRecord
   belongs_to :project
   belongs_to :tag
+  validates_uniqueness_of :project_id, scope: [:tag_id]
```

Comment:
I think this is unnecessary and in fact might break projects in production as they might not have unique tags now. @tachyons might have more inputs.

Reply:
We can have a migration that deletes duplicate tags

Reply:
I don't think it will break the production if there are duplicates already(I tested for that) but we can still have a migration. I can even remove it if you guys wish.

Reply:
I manually added 2 duplicate tags and ran the code on my localhost and there was no problem so I think it will be alright if there are pre-existing duplicate tags.
![5](https://user-images.githubusercontent.com/32474302/55236697-8a273080-5256-11e9-8056-75b41d7d5b09.gif)

Reply:
Write the migration. Make sure you make changes only to tags and not anything else.